src/v/s3/client.h

```diff
@@ -223,6 +222,34 @@ class client {
       const object_key& key,
       const ss::lowres_clock::duration& timeout);
 
+    struct delete_objects_result {
+        struct key_reason {
+            object_key key;
+            ss::sstring reason;
+        };
+        std::vector<key_reason> undeleted_keys{};
+    };
+
+    constexpr static auto delete_objects_max_keys = 1000;
+
+    /// DeleteObjects request.
+    ///
+    /// \pre the number of keys is <= s3::client::delete_objects_max_keys (as
+    /// per the S3 api)
+    ///
+    /// \param bucket the name of the bucket
+    /// \param keys the keys in the bucket to delete
+    /// \param timeout request timeout
+    /// \return a future of delete_objects_result. This contains
+    /// a list of keys that where not deleted, each with the Error.Code returned
+    /// from the server
+    /// note that keys not found in the bucket are NOT reported (they are
+    /// considered as successfully deleted)
+    ss::future<delete_objects_result> delete_objects(
```

Comment:
Google Cloud Storage does not implement this API, right? If that's the case can we at least update the comment to list this limitation. Once #7706 is merged, we can override this with an implementation that performs object-wise deletes.